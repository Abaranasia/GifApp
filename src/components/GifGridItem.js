import React from 'react'
import PropTypes from 'prop-types';

export default function GifGridItem({title, url}) {
/*** 
 * 
 * Componente que imprime en pantalla cada imagen recibida desde GifGrid
 *  
 * ***/ 

    return (
        <figure  className="foto">
            <a href={url} target="_blank" rel="noreferrer">
                <img src= {url} alt= {title}/>
            </a>
            <figcaption>{title}</figcaption>
        </figure>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
