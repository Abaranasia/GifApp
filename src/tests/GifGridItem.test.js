import React from "react";
import { shallow } from "enzyme";
import GifGridItem from '../components/GifGridItem';

describe('Pruebas del componente GifGridItem', () => {

    const title="titulo";
    const url="https://locahost/foto.jpg";
    const wrapper= shallow(<GifGridItem title={ title } url= { url } />); 


    test('Debe mostrar el componente correctamente', () => {
        
        expect(wrapper).toMatchSnapshot()
    });


    test('Debe tener un figurecaption con el texto', () => {

        const fig= wrapper.find('figcaption');
        expect(fig.text().trim()).toBe( title )
    });


    test('Debe tener la imagen con src= url y alt= title', () => {

        const img= wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title )
    });


    test('La figura debe pertenecer a la clase foto', () => {

        const figura= wrapper.find('figure');
        expect(figura.prop('className').includes('foto')).toBe( true )
    });
});