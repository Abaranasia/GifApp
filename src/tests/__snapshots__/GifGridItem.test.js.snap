// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGridItem Debe mostrar el componente correctamente 1`] = `
<figure
  className="foto"
>
  <a
    href="https://locahost/foto.jpg"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt="titulo"
      src="https://locahost/foto.jpg"
    />
  </a>
  <figcaption>
    titulo
  </figcaption>
</figure>
`;
